### ISSUE TITLE
OutOfMemoryError : Loading large Epub File.

### ISSUE THREAD

**Issue by gmeral** [FolioReader/FolioReader-Android#75]
FolioReader has been working fine so far for me, thanks  a lot !
However when i tried to open a large epub file (~250MB), i got an Out of Memory error.
It seems like everything is loaded in memory.
Does folioReader not support large files or did i miss something ?

Thanks in advance for your help.
```
Throwing OutOfMemoryError "Failed to allocate a 248544534 byte allocation with 16777120 free bytes and 112MB until OOM"
FATAL EXCEPTION: Thread-20338
                                                                              
                                                                                java.lang.OutOfMemoryError: Failed to allocate a 248544534 byte allocation with 16777120 free bytes and 112MB until OOM
                                                                                    at com.folioreader.util.FileUtil.saveTempEpubFile(FileUtil.java:105)
                                                                                    at com.folioreader.util.FileUtil.saveEpubFile(FileUtil.java:39)
                                                                                    at com.folioreader.activity.FolioActivity$3.run(FolioActivity.java:168)
                                                                                    at java.lang.Thread.run(Thread.java:818)
android.view.WindowLeaked: Activity com.folioreader.activity.FolioActivity has leaked window com.android.internal.policy.impl.PhoneWindow$DecorView{2e2134e5 V.E..... R......D 0,0-1002,1845} that was originally added here
                                                                                   at android.view.ViewRootImpl.<init>(ViewRootImpl.java:457)
                                                                                   at android.view.WindowManagerGlobal.addView(WindowManagerGlobal.java:277)
                                                                                   at android.view.WindowManagerImpl.addView(WindowManagerImpl.java:69)
                                                                                   at android.app.Dialog.show(Dialog.java:312)
                                                                                   at com.folioreader.util.ProgressDialog.show(ProgressDialog.java:18)
                                                                                   at com.folioreader.activity.FolioActivity.initBook(FolioActivity.java:163)
                                                                                   at com.folioreader.activity.FolioActivity.onCreate(FolioActivity.java:124)
                                                                                   at android.app.Activity.performCreate(Activity.java:6221)
                                                                                   at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1119)
                                                                                   at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2611)
                                                                                   at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2723)
                                                                                   at android.app.ActivityThread.access$900(ActivityThread.java:172)
                                                                                   at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1422)
                                                                                   at android.os.Handler.dispatchMessage(Handler.java:102)
                                                                                   at android.os.Looper.loop(Looper.java:145)
                                                                                   at android.app.ActivityThread.main(ActivityThread.java:5832)
                                                                                   at java.lang.reflect.Method.invoke(Native Method)
                                                                                   at java.lang.reflect.Method.invoke(Method.java:372)
                                                                                   at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1399)
                                                                                   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1194)
```

**Comment by gmeral** [comment]
The problem in this case seem to come from this line in FileUtil:saveTempEpubFile : 

```
       byte[] bytes = new byte[inputStream.available()];
```

According to https://developer.android.com/reference/java/io/InputStream.html#available()

Note that while some implementations of InputStream will return the total number of bytes in the stream, many will not. It is never correct to use the return value of this method to allocate a buffer intended to hold all data in this stream.

**Comment by mahavir155** [comment]
Should be fixed with v0.3.1 @gmeral ..
